partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java

```diff
@@ -289,5 +297,99 @@ public Long extract(ConnectRecord<?> record) {
         throw new PartitionException("Error encoding partition.");
       }
     }
+
+    private Object getField(Struct struct, String fieldName) {
+      Object field;
+      try {
+        field = struct.get(fieldName);
+      } catch (DataException e) {
+        throw new DataException(
+                String.format("The field named '%s' does not exist.", fieldName), e);
+      }
+      return field;
+    }
+
+    private Object getNestedFieldValue(Struct struct) {
+      final String[] fieldNames = fieldName.split("\\.");
+      Struct tmpStruct = struct;
+      Object tmpObject;
+      try {
+        // Iterate down to final struct
+        int i = 0;
+        while (i < fieldNames.length - 1) {
+          try {
+            tmpObject = getField(tmpStruct, fieldNames[i]);
+          } catch (DataException e) {
+            throw new DataException(
+                    String.format("Unable to find nested field '%s'", fieldNames[i]));
+          }
+          tmpStruct = (Struct) tmpObject;
+          i++;
+        }
+        // Extract from final struct
+        tmpObject = getField(tmpStruct, fieldNames[i]);
+      } catch (DataException e) {
+        throw new DataException(
+                String.format("The nested field named '%s' does not exist.", fieldName), e);
+      }
+      return tmpObject;
```

Comment:
Do you want to return a null tmpObject in the event that fieldNames is an empty array? Or if fieldNames length is 1? Currently if fieldNames.length = 1 it should not iterate...

Reply:
If fieldNames.length is 0, it will throw an exception. if fieldNames.length is 1, it will just grab the field from the first layer.